### ISSUE TITLE
Unable to compute dynamic probabilities with Jointlcmm after using the TimeDepVar command

### ISSUE THREAD

**Issue by OJay92** [CecileProust-Lima/lcmm#11]
In the following code, we first specify a joint model by using the Jointlcmm function. Afterwards we take a small sampleset for which we want to compute dynamic probabilities. 

When fitting this model without the "Var1" as TimeDepVAR (time dependent variable indicating an intermediate time corresponding to a change in the risk of event; equals end of the course if not present) it can compute these without any problems.. 

However when including this command accordingly it becomes impossible to compute the predictions as we keep on receiving the same error: 

> errror in dynpred(joint_fit, sampleset, landmark = c(weeks), var.time = "Time",  : 
>   object 'Xevt' not found

Not sure what goes wrong here actually.. is it the variable it self, something with the coding or maybe construction of the data itself (small fictive subset attached)
[jmset_mini.zip](https://github.com/CecileProust-Lima/lcmm/files/2483593/jmset_mini.zip)
Thank you in advance :)

```
# Fit the joint model timedepvar with  the indication the moment of risk during treatment
joint_fit <- Jointlcmm(Marker ~ poly(Time, degree = 1, raw = FALSE),
                       random = ~ poly(Time, degree = 1, raw = FALSE), TimeDepVar = "Var1",
                       survival = Surv(Time_to_Event, Event) ~ Marker + Treat + Var1, 
                       hazard = "Weibull", subject = "ID", data = jmset_mini,  ng= 1, verbose = FALSE)


# Take a small sample set
sampleset <- jmset_mini[jmset_mini$ID == 7,]
rownames(sampleset) <- NULL
sampleset <- sampleset[,c("ID", "Time", "Marker", "Treat", "Var1",
                          "Event", "Time_to_Event")]

# Specify which weeks we want to see
y <- seq(1:5)
weeks <- sampleset[y,'Time']

# Spit out the dynamic predictions and plug them into a dataframe
dynp_plot1 <- dynpred(joint_fit, sampleset, landmark = c(weeks),
                      var.time = "Time",
                      horizon = c(12), draws = F) # predicting 12 weeks ahead

```

**Comment by CecileProust-Lima** [comment]
Thank you very much for your feedback. 
You are right, there seems to be a bug in the function in the case of a timedep var. We will come back to you when we have fixed it. 
Sorry for the inconvenience. 
Cécile

**Comment by VivianePhilipps** [comment]
The bug in the dynpred function has been fixed.
The current version now enables dynamic predictions with a time-dependent variable.
Thank you!
Viviane

**Comment by OJay92** [comment]
Thanks you both a lot for your help :)

**Comment by VivianePhilipps** [comment]
I did not get this error.
Please note that you have to build and re-install the package each time the code changes, even if the changes only affect the R code. Sourcing the R function usually leads to such errors.

**Comment by OJay92** [comment]
Yes indeed you are right, it works perfectly now.. Thanks a lot for your help and support!
